### ISSUE TITLE
[BUG] Missing types in multiple objects

### ISSUE THREAD

**Issue by mhagendoorn-9292** [TOMP-WG/TOMP-API#345]
## API Version
1.1.0

## Summary
Hi. The following types are missing in the .yaml file

- booking.Pricing
- conditionReturnArea.ReturnArea
- journalEntry.Details
- journalEntry.invoiceId (already fixed in #343)

**Comment by edwinvandenbelt** [comment]
Hello Mike, 

please clarify. In the booking object the field 'pricing' is of type Fare:
           pricing:
              description: The pricing information of the overall booking, in addition to any leg pricing, if not all legs have pricing the 
                                  booking should have the fare
              allOf:
                - $ref: "#/components/schemas/fare"

In conditionReturnArea the field 'returnArea' is of type geojsonPolygon:
            returnArea:
              description: area in which the asset should be returned as GeoJSON Polygon coordinates
              allOf:
                - $ref: "#/components/schemas/geojsonPolygon"

In the journalEntry the field 'details' is or a Fare object or an extraCosts object. 
            details:
              description: the specification of the amount; how is it composed.
              oneOf:
                - $ref: "#/components/schemas/fare"
                - $ref: "#/components/schemas/extraCosts"

Did I miss something?
Regards, Edwin

**Comment by mhagendoorn-9292** [comment]
Hi Edwin,

I came up onto this issue when using the autogenerated code that was generated by [Swagger](https://editor.swagger.io/). If I take `conditionReturnArea` as an example, we see this in the yaml:

```
conditionReturnArea:
      description: a return area. In the condition list there can be multiple return area's.
      allOf:
        - $ref: "#/components/schemas/condition"
        - type: object
          properties:
            stationId:
              description: station to which the asset should be returned
              type: string
            returnArea:
              description: area in which the asset should be returned as GeoJSON Polygon coordinates
              allOf:
                - $ref: "#/components/schemas/geojsonPolygon"
            coordinates:
              $ref: "#/components/schemas/coordinates"
            returnHours:
              description: the return hours of the facility (if different from operating-hours)
              type: array
              items:
                $ref: "#/components/schemas/systemHours"
```

The autogenerator generates the following properties in C# (comments omitted for readability):
```
    [DataMember(Name="stationId", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "stationId")]
    public string StationId { get; set; }

    [DataMember(Name="returnArea", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "returnArea")]
    public   ReturnArea { get; set; }

    [DataMember(Name="coordinates", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "coordinates")]
    public Coordinates Coordinates { get; set; }

    [DataMember(Name="returnHours", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "returnHours")]
    public List<SystemHours> ReturnHours { get; set; }
```

As you can see, it places to whitespaces at the `ReturnArea` property. However, it correctly generated the `Coordinates` field. The difference between those fields in the yaml, as far as I can see, is the fact that the `ReturnArea` field has it's reference in the `allOf` field, wehere the `Coordinate` field does not. 

I'm not fluent enough in the OAS syntax to say if this is the actual problem. How this looks to me, is that the reference shouldn't be in the `allOf` field, but in the root of that object instead.

**Comment by edwinvandenbelt** [comment]
Mike, this seems a problem in the C# generator, not in the specification. This construct of using 'allOf' is only used to add a description above a reference. The OAS doesn't allow it without the "allOf" construct.... To be honest, I think this is an issue in the OAS, but we have to live with it for now. You can generate the correct C# code by removing (the description and) allOf.
e.g. 
          returnArea:
              description: area in which the asset should be returned as GeoJSON Polygon coordinates
              allOf:
                - $ref: "#/components/schemas/geojsonPolygon"

should become:
          returnArea:
              $ref: "#/components/schemas/geojsonPolygon"

If you generate it again, you'll find the correct C# code. I am not aware if other generators do have the same problem.

**Comment by edwinvandenbelt** [comment]
it looks like this is solved in OAS 3.1: https://github.com/OAI/OpenAPI-Specification/issues/1514

**Comment by mhagendoorn-9292** [comment]
Thanks for this information! I did not know this. I'll close this issue now, since this isn't a actual issue.
